compression.go

```diff
@@ -0,0 +1,71 @@
+package kafka
+
+const (
+	CompressionNone int8 = iota
+	CompressionGZIP
+	CompressionSnappy
+	CompressionLZ4
+)
+
+var codecs map[int8]CompressionCodec
+
+// RegisterCompressionCodec registers a compression codec so it can be used by a Writer.
+func RegisterCompressionCodec(codec func() CompressionCodec) {
+	if codecs == nil {
+		codecs = make(map[int8]CompressionCodec)
+	}
+
+	c := codec()
+	codecs[c.Code()] = c
+}
+
+// CompressionCodec represents a compression codec to encode and decode
+// the messages.
+// See : https://cwiki.apache.org/confluence/display/KAFKA/Compression
+type CompressionCodec interface {
+	Code() int8
+	Encode(dst, src []byte) (int, error)
+	Decode(dst, src []byte) (int, error)
```

Comment:
Those methods would deserve to be documented to define how valid compression codec implementations must behave. For example, explaining the error that must be returned if the destination buffer is too small, or what the value returned by the `Code` function should be.